dali/pipeline/operators/op_spec.h

```diff
@@ -32,6 +33,65 @@
 
 namespace dali {
 
+namespace detail {
+
+template <typename T, typename S>
+void copy_vector(std::vector<T> &out, const std::vector<S> &in) {
+  out.reserve(in.size());
+  out.clear();
+  for (auto &v : in) {
+    out.emplace_back(v);
+  }
+}
+
+template <typename T>
+void copy_vector(std::vector<T> &out, const std::vector<bool> &in) {
+  out.reserve(in.size());
+  out.clear();
+  for (auto b : in) {
+    out.emplace_back(b);
+  }
+}
+
+/** @brief This overload simply forwrads the reference */
+template <typename T>
+std::vector<T> &&convert_vector(std::vector<T> &&v) {
```

Comment:
you can use perfect forwarding and get rid of the overload below.
Just change `std::move` by `std::forward<T>`

Reply:
I could if I was forwarding `T`, not `vector<T>`.

Reply:
You are right :)